pdns/recursordist/pdns_recursor.cc

```diff
@@ -327,16 +327,17 @@ LWResult::Result arecvfrom(PacketBuffer& packet, int /* flags */, const ComboAdd
   pident->type = qtype;
   pident->remote = fromaddr;
 
-  int ret = MT->waitEvent(pident, &packet, g_networkTimeoutMsec, now);
+  int ret = MT->waitEvent(pident, &packet, g_networkTimeoutMsec, &now);
```

Comment:
It might make sense to set `len = 0;` here if we want to be consistent, as we are now setting it to `0` in the `LWResult::Result::PermanentError` case but not in the `LWResult::Result::Timeout` one, if I'm not mistaken.